### ISSUE TITLE
Ugly ArgumentOutOfRange if you try to start the game with no map selected

### ISSUE THREAD

**Issue by bookuha** [ForNeVeR/bloodmasters#37]
![image](https://github.com/ForNeVeR/bloodmasters/assets/82080872/011913d0-2db9-4977-b9ed-a577cd280cba)
![image](https://github.com/ForNeVeR/bloodmasters/assets/82080872/7fdd81d2-42d2-4a54-b08b-b397365dbdc0)

I personally think, that it is better to inform the user about the problem in some prettier way.

**Comment by moldyharriet** [comment]
I can look into this, and hopefully utilize some sort of internal dialogue menu to display the error.

**Comment by bookuha** [comment]
> Mind if I look into this?

Of course! Contact me here if you need any help

**Comment by ForNeVeR** [comment]
Yep, absolutely, feel free to think how to improve this. Perhaps let's just show an error message instead of throwing an exception :)

**Comment by moldyharriet** [comment]
@bookuha Hey there, just out of sheer curiosity for my own personal sake.. (going to Uni for CompSci in about 3 weeks)

What IDE do you all use?
I've been bouncing between VS and VSCode.. Just trying to find something I like that allows for text editing, syntax highlighting, source control, and management of a whole project, not just individual files. 

Thanks!

**Comment by bookuha** [comment]
@moldyharriet Afaik we all use Jetbrains Rider

**Comment by ForNeVeR** [comment]
This should not matter (we want to support all the development tools, and prefer approaches that work universally, such as `launchSettings.json`, MSBuild build system etc.).

But, as a disclaimer, I am a platform developer in JetBrains Rider, and thus prefer to use it when working on personal projects. But I have no objections against people VS or VSCode as well, and would be happy to accept contributions and/or help with dev environment setup to users of and other tools, provided they meet the basic quality standards.

**Comment by moldyharriet** [comment]
Yep, this is totally logical, and I appreciate your welcoming efforts. 

I am using VSCode, for convenience sake, and because it is what I am familiar with, but I was asking simply as a personal question, for my own peace of mind, because I'm still relatively new to the development world, and this is my first "major" contribution to any projects of this significance.

**Comment by moldyharriet** [comment]
Yeah, I'm going to have to drop out on this one. I've been banging my head against the wall for about 4 hours straight now trying to get this set up to use namespaces to reduce clutter. I know what I want to do to resolve the issue, and it's pretty quick and easy, but namespaces are killing me, and I can't make them work. \o/

I have absolutely no idea why I'm getting the errors that I am, but they are most definitely my fault, and not on the side of you guys' code. 

I appreciate you letting me try my hand at it nonetheless!
